src/vivarium_public_health/results/mortality.py

```diff
@@ -147,34 +145,34 @@ def calculate_ylls(self, x: pd.DataFrame) -> float:
         died_of_cause = x["exit_time"] > self.clock()
         return x.loc[died_of_cause, "years_of_life_lost"].sum()
 
-    ##################
-    # Report methods #
-    ##################
+    ##############################
+    # Results formatting methods #
+    ##############################
 
-    def formatter(
-        self,
-        measure: str,
-        results: pd.DataFrame,
-    ) -> pd.DataFrame:
+    def format(self, measure: str, results: pd.DataFrame) -> pd.DataFrame:
         results = results.reset_index()
-
         if self.config.aggregate:
             results[COLUMNS.ENTITY] = "all_causes"
         else:
             results.rename(columns={"cause_of_death": COLUMNS.ENTITY}, inplace=True)
+        return results[results[COLUMNS.ENTITY] != "not_dead"]
 
-        results = results[results[COLUMNS.ENTITY] != "not_dead"]
-        results[COLUMNS.MEASURE] = measure
-        results[COLUMNS.ENTITY_TYPE] = "cause"
-        results.loc[
-            results[COLUMNS.ENTITY] == "other_causes", COLUMNS.SUB_ENTITY
-        ] = "other_causes"
-        results.loc[
-            results[COLUMNS.ENTITY] == "all_causes", COLUMNS.SUB_ENTITY
-        ] = "all_causes"
+    def get_entity_type_col(self, measure: str, results: pd.DataFrame) -> pd.Series:
+        values = pd.Series("cause", index=results.index)
```

Comment:
You should be able to use the same approach I suggested in the disability observer here and in `get_sub_entity`.